src/views/BlogDetails.tsx

```diff
@@ -1,7 +1,25 @@
+/** @jsxImportSource @emotion/react */
+import { useParams } from 'react-router-dom';
+import { blogPosts } from 'data';
+import {
+  postContainer,
+  postTitle,
+  postDescription,
+  postText,
+} from './BlogDetails.styles';
+
 export const BlogDetails: React.FC = () => {
+  const params = useParams();
+
   return (
     <div>
-      <p>TODO</p>
+      {params.id && (
+        <div css={postContainer}>
+          <p css={postTitle}>{blogPosts[params.id].title}</p>
+          <p css={postDescription}>{blogPosts[params.id].description}</p>
+          <p css={postText}>{blogPosts[params.id].text}</p>
+        </div>
+      )}
     </div>
```

Comment:
Ovo treba ispraviti: radi ti samo zato što se poklopilo da su id-evi od blog postova 0,1,2 i 3.
Da su id-evi neki drugi brojevi npr. 10, pokusala bi pristupit blog post arrayu sa ne postojecim indexom i aplikacija bi ti pukla. 

Sa _useParams_ hookom mozes odmah izvuci id ( pogledaj ovaj primjer + prouci i object destructuring u typescriptu: https://reactrouter.com/docs/en/v6/hooks/use-params ) pa probaj skuziti kako da iz blogPosts array-a izvuces blog s tim id-em.